pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -84,20 +82,25 @@ func New(context *clusterd.Context, clusterInfo *cephclient.ClusterInfo, spec ce
 		clusterInfo: clusterInfo,
 		spec:        spec,
 		rookVersion: rookVersion,
-		Replicas:    1,
+		Replicas:    spec.Mgr.Count,
 		exitCode:    exec.ExitStatus,
 	}
 }
 
 var updateDeploymentAndWait = mon.UpdateCephDeploymentAndWait
 
+// for backward compatibility, default to 1 mgr
+func (c *Cluster) getReplicas() int {
+	replicas := c.Replicas
+	if replicas == 0 {
+		replicas = 1
+	}
+	return replicas
+}
+
 func (c *Cluster) getDaemonIDs() []string {
 	var daemonIDs []string
-	for i := 0; i < c.Replicas; i++ {
-		if i >= 2 {
-			logger.Errorf("cannot have more than 2 mgrs")
-			break
-		}
+	for i := 0; i < c.getReplicas(); i++ {
```

Comment:
Should we print a small warning if `i == 2`? Like mentioned earlier `Set this number only when multiple ceph managers are needed. Due to mgr's mechanism, plural mgrs might cause issues # when accessing dashboard through service cluster IP.`

Also, should we fail if `i > 2`?

Reply:
I updated the comment you mentioned in cluster.yaml about accessing the dashboard. That issue is fixed now.

> Also, should we fail if i > 2?

The crd schema only allows a max of two. We really shouldn't hit this condition, so logging it seems fine.

Reply:
I just guess I wasn't sure if the CRD definition only was working as expected but if you tell me it does :)